src/hooks/useFormattedDate.js

```diff
@@ -0,0 +1,14 @@
+import { format } from 'date-fns';
+
+const useFormattedDate = (date, formatString = 'MMMM d, yyyy') => {
+  if (!date) return 'Unknown Date';
```

Comment:
You should return null in case of an invalid date

The UI error should be handled in other places